TicketUtil.py

```diff
@@ -119,6 +120,207 @@ def close_requests_session(self):
+            r.raise_for_status()
+            resp = r.json()
+            self.token = resp['token']
+            logging.debug("Create requests session: Status Code: {0}".format(r.status_code))
+            logging.info("Successfully authenticated to {0} with token: {1}".format(self.ticketing_tool, self.token))
+            return s
+
+        # We log an error if authentication was not successful, because rest of the HTTP requests will not succeed.
+        except requests.RequestException as e:
+            logging.error("Error authenticating to {0}. No valid credentials were provided.".format(self.auth_url))
+            logging.error(e.args[0])
+
+
+    def create_ticket(self, params):
+        """
+        Tries to create the ticket through the ticketing tool's API.
+        Retrieves the ticket_id and creates the ticket_url.
+        :param params: The payload to send in the POST request.
+        :return:
+        """
+        # Attempt to create ticket.
+        try:
+
+            if self.token:
+                params['token'] = self.token
+
+            r = self.s.post(self.rest_url, params=params)
+
+            r.raise_for_status()
+            logging.debug("Create ticket: Status Code: {0}".format(r.status_code))
+
+
+            self.ticket_id = r.json()['id']
+            self.ticket_url = self.craft_ticket_url()
+            logging.info("Created ticket {0} - {1}".format(self.ticket_id, self.ticket_url))
+
+        # If ticket creation is not successful, log an error.
+        except requests.RequestException as e:
+            logging.error("Error creating ticket")
+            logging.error(e.args[0])
+
+
+    def add_comment(self, comment):
+        """
+        Adds a comment to a Bugzilla ticket.
+        :param comment: A string representing the comment to be added.
+        :return:
+        """
+        if not self.ticket_id: # Create the payload for our update.
+            logging.error("No ticket ID associated with ticket object. Set ticket ID with set_ticket_id(ticket_id)")
+            return
+
+        try:
+            params = {'comment': comment}
+
+            if self.token:
+                params['token'] = self.token
+            r = self.s.post("{0}/{1}/comment".format(self.rest_url, self.ticket_id), json=params)
+            r.raise_for_status()
+            logging.debug("Add comment: Status Code: {0}".format(r.status_code))
+            logging.info("Updated ticket {0} - {1}".format(self.ticket_id, self.ticket_url))
+            # Instead of using e.message, use e.args[0] instead to prevent DeprecationWarning for exception.message.
+
+        except requests.RequestException as e:
+            logging.error("Error updating ticket")
+            logging.error(e.args[0])
+
+    def transition_ticket(self, resolve_params):
+        """
+        Resolving  a Bugzilla ticket.
+        :param resolution: A string representing the resolution to be added.
```

Comment:
The parameter for this method is named resolve_params. This docstring says "resolution", when it should say "resolve_params"